.github/workflows/build.yaml

```
@@ -10,20 +10,38 @@ on:
 
 jobs:
   build-latest:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         compiler: [cc, clang, clang-12, gcc-10]
+        os: [ubuntu-latest, macos-latest]
+        exclude:
+          - os: macos-latest
+            compiler: cc
+          - os: macos-latest
+            compiler: clang-12
+          - os: macos-latest
+            compiler: gcc-10
     steps:
       - uses: actions/checkout@v3
+      - name: install macOS autogen prerequisites
+        run: brew install autoconf automake libtool
+        if: runner.os == 'macOS'
       - name: autogen
         run: ./autogen.sh
       - name: compiler
         run: $CC -v
         env:
           CC: ${{ matrix.compiler }}
-      - name: install missing deps
+      - name: install missing Linux deps
         run: sudo apt-get install libtls-dev
+        if: runner.os == 'Linux'
+      - name: install missing macOS deps
+        run: |
+          brew install libretls openssl@3 rsync
+          echo "LDFLAGS=-L/usr/local/opt/openssl@3/lib" >> $GITHUB_ENV
+          echo "CPPFLAGS=-I/usr/local/opt/openssl@3/include" >> $GITHUB_ENV
```

You can also do this if you don't want to hardcode the prefix.
```suggestion
          echo "LDFLAGS=-L$(brew --prefix openssl@3)/lib" >> $GITHUB_ENV
          echo "CPPFLAGS=-I$(brew --prefix openssl@3)/include" >> $GITHUB_ENV
```

---

For consistency, this makes absolutely sense – applied.